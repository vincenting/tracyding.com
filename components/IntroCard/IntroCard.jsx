import dynamic from 'next/dynamic'
import React, { useEffect, useRef } from 'react'

const WelcomeAnimation = dynamic(() => import('../WelcomeAnimation/WelcomeAnimation'), {
  ssr: false
})

export default function IntroCard (props) {
  const audio = useRef()
  useEffect(() => {
    audio.current = new Audio('/audio/hi_im_tracy.mp3')
  }, [])

  const playHelloFromTracy = () => {
    if (audio.current) {
      audio.current.currentTime = 0
      audio.current.play()
    }
  }

  return (
    <>
      <section {...props}>
        <div className='w-fit ml-auto mr-auto'>
          <WelcomeAnimation />
          <h3 className='text-lg text-center'>
              Dear visitor,
              I&apos;m <span onClick={playHelloFromTracy} className='text-pink-500 underline cursor-pointer'>Tracy.</span> ðŸ˜Š
          </h3>
          <p className='mt-2'>My Portfolio Website is coming soon. </p>
        </div>
      </section>
    </>
  )
}
