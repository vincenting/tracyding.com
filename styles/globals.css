@tailwind base;
@tailwind components;
@tailwind utilities;

@font-face {
    font-family: "baby_doll";
    src: url("/fonts/baby_doll.woff2") format("woff2"), url("/fonts/baby_doll.woff") format("woff");
    font-weight: normal;
    font-style: normal;
}

@supports (-webkit-touch-callout: none) {
    .h-screen {
        /* The hack for Safari */
        height: -webkit-fill-available;
    }
}
