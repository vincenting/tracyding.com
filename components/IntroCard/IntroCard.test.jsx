import { describe, expect, it } from '@jest/globals'
import React from 'react'

import { render } from '@testing-library/react'

import IntroCard from './IntroCard'

describe('IntroCard Page', () => {
  it('should render IntroCard page as expected', () => {
    expect(render(<IntroCard />).baseElement).toMatchSnapshot()
  })
})
