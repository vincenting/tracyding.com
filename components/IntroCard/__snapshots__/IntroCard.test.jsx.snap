// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroCard Page should render IntroCard page as expected 1`] = `
<Fragment>
  <section>
    <div
      className="w-fit ml-auto mr-auto"
    >
      <Component />
      <h3
        className="text-lg text-center"
      >
        Dear visitor, I'm 
        <span
          className="text-pink-500 underline cursor-pointer"
          onClick={[Function]}
        >
          Tracy.
        </span>
         ðŸ˜Š
      </h3>
      <p
        className="mt-2"
      >
        My Portfolio Website is coming soon. 
      </p>
    </div>
  </section>
</Fragment>
`;
